src/lib.rs

```diff
@@ -1276,8 +1248,11 @@ impl Build {
                 // the SDK, but for all released versions of the
                 // Windows SDK it is required.
                 if target.contains("arm") || target.contains("thumb") {
-                    cmd.args
-                        .push("/D_ARM_WINAPI_PARTITION_DESKTOP_SDK_AVAILABLE=1".into());
+                    if self.cuda {
```

Comment:
Are changes like this necessary? This PR feels a bit overreaching in terms of changing everything it finds?

Reply:
I wasn't too sure about this define, so I added it when compiling with nvcc as well. We can just simplify it to a `-D` option instead and it will define it for both nvcc & non-nvcc versions?